From e2c327740cc036ef2e45aabbbc0d48ab67f2e088 Mon Sep 17 00:00:00 2001
From: dianlujitao <dianlujitao@lineageos.org>
Date: Wed, 14 Nov 2018 16:18:32 +0800
Subject: [PATCH 4/4] ProfileManager: Don't start before decrypted

 * WiFi service is not considered as a core service so it won't start in
   min framework, but ProfileManager accesses to WiFi service's API.
   This would lead to a fatal exception and trigger RescueParty.
 * We don't really need ProfileManager during such early stage, so
   simply disable it to avoid invoking WifiManager.

Change-Id: I2c66217599e0062eaff2e5f4e86ebc092d4a41d5
---
 .../lineageos/platform/internal/ProfileManagerService.java   | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/lineage/lib/main/java/org/lineageos/platform/internal/ProfileManagerService.java b/lineage/lib/main/java/org/lineageos/platform/internal/ProfileManagerService.java
index d386b3f..a46c783 100644
--- a/lineage/lib/main/java/org/lineageos/platform/internal/ProfileManagerService.java
+++ b/lineage/lib/main/java/org/lineageos/platform/internal/ProfileManagerService.java
@@ -247,6 +247,11 @@ public class ProfileManagerService extends LineageSystemService {
         return LineageContextConstants.Features.PROFILES;
     }
 
+    @Override
+    public boolean isCoreService() {
+        return false;
+    }
+
     @Override
     public void onStart() {
         mBackupManager = new BackupManager(mContext);
-- 
2.17.1

